lib/api/rule.go

```diff
@@ -27,17 +27,21 @@ type Rule struct {
 	Action string `json:"action" validate:"action"`
 
 	Protocol *Protocol `json:"protocol,omitempty" validate:"omitempty"`
-	ICMPType *int      `json:"icmpType,omitempty" validate:"omitempty,gte=0,lte=255"`
-	ICMPCode *int      `json:"icmpCode,omitempty" validate:"omitempty,gte=0,lte=255"`
+	ICMP     *CodeType `json:"icmp,omitempty" validate:"omitempty"`
 
 	NotProtocol *Protocol `json:"!protocol,omitempty" validate:"omitempty"`
-	NotICMPType *int      `json:"!icmpType,omitempty" validate:"omitempty,gte=0,lte=255"`
-	NotICMPCode *int      `json:"!icmpCode,omitempty" validate:"omitempty,gte=0,lte=255"`
+	NotICMP     *CodeType `json:"!icmp,omitempty" validate:"omitempty"`
 
 	Source      EntityRule `json:"source,omitempty" validate:"omitempty"`
 	Destination EntityRule `json:"destination,omitempty" validate:"omitempty"`
 }
 
+// CodeType defines structure for ICMP and NotICMP sub-struct for ICMP code and type
+type CodeType struct {
```

Comment:
I think a better name for this might be something like `ICMPFields` - I think it captures the intention of the struct a bit better.

@gunjan5 thoughts?